modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1455,17 +1448,19 @@ private void dropTableLocally(long causalityToken, CatalogTableDescriptor tableD
         }
     }
 
-    private Set<Assignment> calculateAssignments(TablePartitionId tablePartitionId) {
+    private CompletableFuture<Set<Assignment>> calculateAssignments(TablePartitionId tablePartitionId) {
         int catalogVersion = catalogService.latestCatalogVersion();
 
         CatalogTableDescriptor tableDescriptor = getTableDescriptor(tablePartitionId.tableId(), catalogVersion);
 
-        return AffinityUtils.calculateAssignmentForPartition(
-                // TODO: https://issues.apache.org/jira/browse/IGNITE-19425 we must use distribution zone keys here
-                baselineMgr.nodes().stream().map(ClusterNode::name).collect(toList()),
-                tablePartitionId.partitionId(),
-                getZoneDescriptor(tableDescriptor, catalogVersion).replicas()
-        );
+        CatalogZoneDescriptor zoneDescriptor = getZoneDescriptor(tableDescriptor, catalogVersion);
+
+        return distributionZoneManager.dataNodes(zoneDescriptor.updateToken(), tableDescriptor.zoneId()).thenApply(dataNodes ->
+                AffinityUtils.calculateAssignmentForPartition(
+                        dataNodes,
+                        tablePartitionId.partitionId(),
+                        getZoneDescriptor(tableDescriptor, catalogVersion).replicas()
```

Comment:
Seems that we already have zoneDesciptor variable, right?

Reply:
Fixed